### PR TITLE
Prepare  transaction_logic for writing tests.

### PR DESCRIPTION
**Pull request by Sventimir**
This PR refactors the `transaction_logic` library n order to facilitate writing unit tests. No semantic changes are expected.

In particular:
* Move signatures from `Zkapp_command_logic` into a separate module to make the implementation file smaller.
* Move the implementation of `Bool` input to `Zkapp_command_logic.Make()` into a sepate module as it does not depend on inputs to `Mina_transaction_logic.Make()`.
* Write generic tests for `Bool_intf` impenetations and apply them to the particular implementation mentioned above (there are other implementations , but applying the functor to them is not easy, at least for the moment).

Explain how you tested your changes:
* This PR just moves the code around without any substantial change (and adds tests). The CI guarantees the system still works.

* This is a step towards #12530 and #12544.

### CODE REVIEW

**Comment by psteckler** [comment]
Nit: In `Zkapp_command_logic.Local_state.Checked.t`, the parameterized type should be `t`, instead of `Stable.Latest.t`.

File: src/lib/transaction_logic/test/transaction_test_lib.ml [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    end)
+
+    let var i bs = Array.get bs i
+
+    let exec ~(var_count : int) ?(expect = true) (test : Logic.t t) =
+      List.iter (make var_count) ~f:(fun bs ->
+          [%test_result: bool] (Logic.eval @@ test bs) ~expect )
+  end
+
+  let%test_module "Logic tests" =
```
Comment (psteckler) [inline src/lib/transaction_logic/test/transaction_test_lib.ml:57]: Not sure these tests are needed here. Surely these properties hold for the OCaml `bool` type. 

For the Snarky type `Boolean.var`, used in `Transaction_snark.Inputs`, wouldn't it be better to have tests in Snarky. The tests seem unrelated to transaction logic as such.

Comment (Sventimir) [reply src/lib/transaction_logic/test/transaction_test_lib.ml:57]: You're probably right. I already tried to instantiate the module for `Boolean.var`, but it's not straightforward, as there's no evaluation function I could find in the scope. I supposse I'd have to import more packages from `snarky`, in which case it's likely better to put these tests directly in `snarky`. I'll discard this commit and write appropriate tests in `snarky`.

Comment (Sventimir) [reply src/lib/transaction_logic/test/transaction_test_lib.ml]: I removed these tests. I'll add those for `snarky` in a separate PR.

Comment (psteckler) [reply src/lib/transaction_logic/test/transaction_test_lib.ml]: I would also suggest that only the truth tables for ||, &&, and NOT would need testing; properties like associativity, De Morgan laws, etc., follow directly from those.

**Comment by Sventimir** [comment]
I'm closing this pull request as we have done more refactorings on another branch, which also includes these changes. The PR is coming soon .
